app/charts/area/areas-state.tsx

```diff
@@ -106,11 +105,13 @@ const useAreasState = ({
     const v = d[fields.y.componentIri];
     return v !== null ? +v : null;
   };
+  const segmentKey = fields.segment ? fields.segment.componentIri : "segment";
   const getSegment = useCallback(
     (d: Observation): string =>
       fields.segment ? (d[fields.segment.componentIri] as string) : "segment",
     [fields.segment]
   );
+  const allSegments = [...new Set(data.map((d) => getSegment(d)))];
```

Comment:
Nit: For the record, I had trouble understand "segment", maybe stackVar would be better ? Not for this PR though.